akka-stream-tests/src/test/scala/akka/stream/scaladsl/MapWithResourceSpec.scala

```diff
@@ -236,16 +236,14 @@ class MapWithResourceSpec extends StreamSpec(UnboundedMailboxConfig) {
     }
 
     "fail when close throws exception" in {
-      val out = TestSubscriber.probe[String]()
+      val sub = TestSubscriber.probe[String]()
       Source
         .single(1)
         .mapWithResource(() => Iterator("a"))((it, _) => if (it.hasNext) Some(it.next()) else None, _ => throw TE(""))
         .collect { case Some(line) => line }
-        .runWith(Sink.fromSubscriber(out))
+        .runWith(Sink.fromSubscriber(sub))
 
-      out.request(61)
-      out.expectNext("a")
-      out.expectError(TE(""))
+      sub.ensureSubscription().request(1).expectNext("a").expectError(TE(""))
```

Comment:
What I was talking to was exception triggered by elements, but in this case the exception is triggered by completion, which also travels in flow, so never overtakes the elements but may happen without any demand, so the race in this test could still be here.

If it is still flaky a solution would be to use a test publisher as well so that you can control the timing of the source completing more explicitly.

Reply:
Let me fix it in this then.

Reply:
I think it should be fine now. @johanandren